### PR TITLE
Commit for #156

### PR DESCRIPTION
**Pull request by mcmilwj**
Added support for API functionality as described at https://box-content.readme.io/reference#move-folder-into-another-users-folder

Commit for #156

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @mcmilwj has signed the CLA. Thanks for the pull request!

File: src/main/java/com/box/sdk/BoxUser.java [hunk 1]
```diff
@@ -308,6 +309,34 @@ public void updateInfo(BoxUser.Info info) {
         JsonObject jsonObject = JsonObject.readFrom(response.getJSON());
         info.update(jsonObject);
     }
+    
+        /**
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxUser.java]: Fix indentation.

File: src/main/java/com/box/sdk/BoxUser.java [hunk 2]
```diff
@@ -308,6 +309,34 @@ public void updateInfo(BoxUser.Info info) {
...
+     * Moves all of the owned content from within one user’s folder into a new folder in another user's account.
+     * You can move folders across users as long as the you have administrative permissions and the 'source' 
+     * user owns the folders. Per the documentation at the link below, this will move everything from the root 
+     * folder, as this is currently the only mode of operation supported.
+     * 
+     * See also https://box-content.readme.io/reference#move-folder-into-another-users-folder
+     * 
+     * @param sourceUserId the user id of the user whose files will be the source for this operation
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxUser.java]: Capitalize ID. This is just a stylistic thing we try to be consistent with this throughout the SDK.

File: src/main/java/com/box/sdk/BoxUser.java [hunk 3]
```diff
@@ -308,6 +309,34 @@ public void updateInfo(BoxUser.Info info) {
...
+     * @return BoxFolder.Info info for the newly created folder
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxUser.java]: I don't think you need the `BoxFolder.Info` type in the return tag.

File: src/main/java/com/box/sdk/BoxUser.java [hunk 4]
```diff
@@ -308,6 +309,34 @@ public void updateInfo(BoxUser.Info info) {
...
+     */
+    public BoxFolder.Info moveFolderToUser(final String sourceUserId) {
+        // Currently the API only supports moving of the root folder (0), hence the hard coded "0"
+        final URL url = MOVE_FOLDER_TO_USER_TEMPLATE.build(this.getAPI().getBaseURL(), sourceUserId, "0");
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxUser.java]: Another stylistic thing - we avoid using `final` on parameters and local variables (except where necessary for closures).

**Comment by gcurtis** [comment]
Thanks for adding this feature! All the logic looks solid, I just left some comments on stylistic things. It looks like the build is failing because of some checkstyle errors:

> :checkstyleMain[ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxUser.java:38: Line has trailing spaces.
> [ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxUser.java:312: Line has trailing spaces.
> [ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxUser.java:315: Line has trailing spaces.
> [ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxUser.java:316: Line has trailing spaces.
> [ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxUser.java:318: Line has trailing spaces.
> [ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxUser.java:320: Line has trailing spaces.
> [ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxUser.java:337: Line has trailing spaces.

You should be able to reproduce those errors locally by running `gradle check`. As soon as you address the comments (and squash the commits) I'll merge this.

**Pull request closed without merging**